### ISSUE TITLE
Windows 10 - Taskbar entry context menu is empty 

### ISSUE THREAD

**Issue by timocov** [microsoft/vscode#121184]
Issue Type: <b>Bug</b>

1. Install VSCode (tried both stable and insiders) for current user
1. Open installed VSCode
1. File -> Open Folder (open several different folders to fill "recent" list)
1. Right click on VSCode's icon on Taskbar

=> There is no recent folders, only task "New Window"

VS Code version: Code - Insiders 1.56.0-insider (48387dfc3d691558404cff1ea2582e3862a40080, 2021-04-13T05:15:20.350Z)
OS version: Windows_NT x64 10.0.19042

Extensions: none

![image](https://user-images.githubusercontent.com/3112183/114567979-68057d00-9c7c-11eb-868d-c4a7ca50d2d9.png)

![image](https://user-images.githubusercontent.com/3112183/114568005-6fc52180-9c7c-11eb-99ea-ddb7422d903d.png)

**Comment by bpasero** [comment]
Please see if you can find any logged output that could help us find the source of the issue:
* run `code --verbose` from the command line and attach the output
* in Code select **Help | Toggle Developer Tools** and select the **Console**
* attach any output you see in the **Console** you can find

Thanks!

**Comment by timocov** [comment]
It seems this is what we're looking for

```
TRACE Error while resolving configuration file 'vscode-userdata:/c%3A/Users/etimokhov/AppData/Roaming/Code%20-%20Insiders/User/settings.json': Unable to read file 'vscode-userdata:/c:/Users/etimokhov/AppData/Roaming/Code - Insiders/User/settings.json' (Error: Unable to resolve non-existing file 'vscode-userdata:/c:/Users/etimokhov/AppData/Roaming/Code - Insiders/User/settings.json')
log.ts:272 TRACE Error while resolving configuration file 'vscode-userdata:/c%3A/Users/etimokhov/AppData/Roaming/Code%20-%20Insiders/User/tasks.json': Unable to read file 'vscode-userdata:/c:/Users/etimokhov/AppData/Roaming/Code - Insiders/User/tasks.json' (Error: Unable to resolve non-existing file 'vscode-userdata:/c:/Users/etimokhov/AppData/Roaming/Code - Insiders/User/tasks.json')
```

**Comment by bpasero** [comment]
Hm, no. That is an error but I think unrelated.

**Comment by timocov** [comment]
Probably the full log from devtools would be better then?

[-1618492618863.log](https://github.com/microsoft/vscode/files/6318280/-1618492618863.log)

Let me know if I can provide you anything else.

**Comment by timocov** [comment]
And here is one from command line

```
[main 2021-04-15T13:16:07.057Z] Starting VS Code
[main 2021-04-15T13:16:07.058Z] from: c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app
[main 2021-04-15T13:16:07.058Z] args: {
  _: [],
  diff: false,
  add: false,
  goto: false,
  'new-window': false,
  'reuse-window': false,
  wait: false,
  help: false,
  'list-extensions': false,
  'show-versions': false,
  version: false,
  verbose: true,
  status: false,
  'prof-startup': false,
  'prof-v8-extensions': false,
  'disable-extensions': false,
  'disable-gpu': false,
  telemetry: false,
  debugRenderer: false,
  logExtensionHostCommunication: false,
  'skip-release-notes': false,
  'disable-telemetry': false,
  'disable-updates': false,
  'disable-keytar': false,
  'disable-crash-reporter': false,
  'crash-reporter-id': '11681bbe-6fa9-4a86-a82f-8eb0c752dc92',
  'skip-add-to-recently-opened': false,
  'unity-launch': false,
  'open-url': false,
  'file-write': false,
  'file-chmod': false,
  'driver-verbose': false,
  force: false,
  'do-not-sync': false,
  trace: false,
  'force-user-env': false,
  'force-disable-user-env': false,
  'open-devtools': false,
  __sandbox: false,
  'no-proxy-server': false,
  nolazy: false,
  'force-renderer-accessibility': false,
  'ignore-certificate-errors': false,
  'allow-insecure-localhost': false,
  logsPath: 'C:\\Users\\etimokhov\\AppData\\Roaming\\Code - Insiders\\logs\\20210415T161607'
}
[main 2021-04-15T13:16:07.063Z] Deleting Chrome cached data path: c:\Users\etimokhov\AppData\Roaming\Code - Insiders\Code Cache
[main 2021-04-15T13:16:07.070Z] Resolving machine identifier...
[main 2021-04-15T13:16:07.070Z] Resolved machine identifier: 296f235928e1b68fc33673c97b16f0858b160bf30d9bc1a25ce864b568b8a91f
[main 2021-04-15T13:16:07.080Z] Main->SharedProcess#connect
[main 2021-04-15T13:16:07.131Z] StorageMainService: creating global storage
[main 2021-04-15T13:16:07.131Z] lifecycle (main): phase changed (value: 2)
[main 2021-04-15T13:16:07.132Z] windowsManager#open
[main 2021-04-15T13:16:07.134Z] windowsManager#open pathsToOpen [
  {
    workspace: { id: 'bf0fb64e91d1d2217f5a316b57076d19', uri: [w] },
    exists: true
  }
]
[main 2021-04-15T13:16:07.136Z] IPC Object URL: Registered new channel vscode:a851c0dd-d5f3-41d7-a7bf-4e0dc21f17d9.
[main 2021-04-15T13:16:07.137Z] window#validateWindowState: validating window state on 2 display(s) { mode: 0, x: 2547, y: 381, width: 1443, height: 768 }
[main 2021-04-15T13:16:07.137Z] window#validateWindowState: multi-monitor working area { x: 2061, y: -255, width: 2419, height: 1440 }
[main 2021-04-15T13:16:07.138Z] window#ctor: using window state { mode: 0, x: 2547, y: 381, width: 1443, height: 768 }
[main 2021-04-15T13:16:07.139Z] window#ctor: vscode-file:// protocol is explicitly disabled
[main 2021-04-15T13:16:07.265Z] StorageMainService: creating workspace storage (bf0fb64e91d1d2217f5a316b57076d19)
[main 2021-04-15T13:16:07.267Z] windowsManager#open used window count 1 (workspacesToOpen: 0, foldersToOpen: 1, emptyToRestore: 0, emptyToOpen: 0)
[main 2021-04-15T13:16:07.270Z] lifecycle (main): phase changed (value: 3)
[main 2021-04-15T13:16:07.290Z] update#setState idle
[main 2021-04-15T13:16:07.292Z] resolveShellEnv(): skipped (Windows)
(node:23624) electron: The default of contextIsolation is deprecated and will be changing from false to true in a future release of Electron.  See https://github.com/electron/electron/issues/23506 for more information
[main 2021-04-15T13:16:07.656Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:07.658Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:07.720Z] resolveShellEnv(): skipped (Windows)
[main 2021-04-15T13:16:07.869Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:08.667Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:08.669Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:08.691Z] SharedProcess: on vscode:createSharedProcessMessageChannel
[main 2021-04-15T13:16:08.691Z] resolveShellEnv(): skipped (Windows)
[main 2021-04-15T13:16:08.692Z] IPC Object URL: Registered new channel vscode:a1814ca1-ec12-4476-b138-1db64367763e.
(node:23624) electron: The default of contextIsolation is deprecated and will be changing from false to true in a future release of Electron.  See https://github.com/electron/electron/issues/23506 for more information
[main 2021-04-15T13:16:08.721Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:08.901Z] resolveShellEnv(): skipped (Windows)
[main 2021-04-15T13:16:08.928Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:09.100Z] RequestService#request https://update.code.visualstudio.com/api/update/win32-x64-user/insider/25785b97e533b15ae3b12e48aaf4187416437e62
[main 2021-04-15T13:16:09.102Z] resolveShellEnv(): skipped (Windows)
[main 2021-04-15T13:16:09.106Z] SharedProcess: IPC ready
[main 2021-04-15T13:16:09.109Z] Main->SharedProcess#connect: connection established
[main 2021-04-15T13:16:09.151Z] SharedProcess: Overall ready
[main 2021-04-15T13:16:09.597Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:09.598Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:09.607Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:09.807Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:16:37.292Z] update#checkForUpdates, state =  idle
[main 2021-04-15T13:16:37.292Z] update#setState checking for updates
[main 2021-04-15T13:16:37.292Z] RequestService#request https://update.code.visualstudio.com/api/update/win32-x64-user/insider/25785b97e533b15ae3b12e48aaf4187416437e62
[main 2021-04-15T13:16:37.294Z] resolveShellEnv(): skipped (Windows)
[main 2021-04-15T13:16:37.375Z] update#setState idle
[main 2021-04-15T13:17:03.401Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:17:03.403Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:17:03.610Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:18:08.527Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:18:08.528Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:18:08.741Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:19:08.519Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:19:08.521Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:19:08.736Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:21:08.549Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:21:08.553Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
[main 2021-04-15T13:21:08.760Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Roaming\Code - Insiders\User\globalStorage
```

**Comment by bpasero** [comment]
Can you try to run `code --disable-extensions --user-data-dir <directory>` where `<directory>` is an empty folder? This will ensure Code is starting with a fresh data directory, e.g. no specific settings and without any extensions running.

**Comment by timocov** [comment]
Although it is a fresh installed version of VSCode Insiders (I especially installed it to check whether the issue is fixed in the next version) I can do that. I run it like this `./Code\ -\ Insiders.exe --disable-extensions --user-data-dir ./test --verbose` (in the folder with installed vscode) and the issue still persist. Here are logs from this run if you need them:

From devtools: [-1618556102341.log](https://github.com/microsoft/vscode/files/6323095/-1618556102341.log)

From console:

```
[main 2021-04-16T06:54:42.655Z] Starting VS Code
[main 2021-04-16T06:54:42.655Z] from: c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\resources\app
[main 2021-04-16T06:54:42.656Z] args: {
  _: [],
  diff: false,
  add: false,
  goto: false,
  'new-window': false,
  'reuse-window': false,
  wait: false,
  'user-data-dir': './test',
  help: false,
  'list-extensions': false,
  'show-versions': false,
  version: false,
  verbose: true,
  status: false,
  'prof-startup': false,
  'prof-v8-extensions': false,
  'disable-extensions': true,
  'disable-gpu': false,
  telemetry: false,
  debugRenderer: false,
  logExtensionHostCommunication: false,
  'skip-release-notes': false,
  'disable-telemetry': false,
  'disable-updates': false,
  'disable-keytar': false,
  'disable-crash-reporter': false,
  'crash-reporter-id': '11681bbe-6fa9-4a86-a82f-8eb0c752dc92',
  'skip-add-to-recently-opened': false,
  'unity-launch': false,
  'open-url': false,
  'file-write': false,
  'file-chmod': false,
  'driver-verbose': false,
  force: false,
  'do-not-sync': false,
  trace: false,
  'force-user-env': false,
  'force-disable-user-env': false,
  'open-devtools': false,
  __sandbox: false,
  'no-proxy-server': false,
  nolazy: false,
  'force-renderer-accessibility': false,
  'ignore-certificate-errors': false,
  'allow-insecure-localhost': false,
  logsPath: 'C:\\Users\\etimokhov\\AppData\\Local\\Programs\\Microsoft VS Code Insiders\\test\\logs\\20210416T095442'
}
[main 2021-04-16T06:54:42.660Z] Deleting Chrome cached data path: c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\Code Cache
[main 2021-04-16T06:54:42.675Z] Resolving machine identifier...
[main 2021-04-16T06:54:42.675Z] Resolved machine identifier: 0d9ee7d293f9207e490bf6b0b7924aadbbfbeaae00a0ffbc7c716ed02e2d2436
[main 2021-04-16T06:54:42.681Z] Main->SharedProcess#connect
[main 2021-04-16T06:54:42.700Z] StorageMainService: creating global storage
[main 2021-04-16T06:54:42.700Z] lifecycle (main): phase changed (value: 2)
[main 2021-04-16T06:54:42.701Z] windowsManager#open
[main 2021-04-16T06:54:42.703Z] windowsManager#open pathsToOpen [
  {
    workspace: { id: 'bf0fb64e91d1d2217f5a316b57076d19', uri: [w] },
    exists: true
  }
]
[main 2021-04-16T06:54:42.706Z] IPC Object URL: Registered new channel vscode:b9840501-6e4f-432f-b5c7-e7c055fc8b09.
[main 2021-04-16T06:54:42.707Z] window#validateWindowState: validating window state on 2 display(s) { mode: 1, x: 448, y: 216, width: 1024, height: 768 }
[main 2021-04-16T06:54:42.708Z] window#validateWindowState: multi-monitor working area { x: 127, y: 0, width: 1793, height: 1200 }
[main 2021-04-16T06:54:42.709Z] window#ctor: using window state { mode: 1, x: 448, y: 216, width: 1024, height: 768 }
[main 2021-04-16T06:54:42.709Z] window#ctor: vscode-file:// protocol is explicitly disabled
[main 2021-04-16T06:54:42.785Z] StorageMainService: creating workspace storage (bf0fb64e91d1d2217f5a316b57076d19)
[main 2021-04-16T06:54:42.786Z] windowsManager#open used window count 1 (workspacesToOpen: 0, foldersToOpen: 1, emptyToRestore: 0, emptyToOpen: 0)
[main 2021-04-16T06:54:42.787Z] lifecycle (main): phase changed (value: 3)
[main 2021-04-16T06:54:42.810Z] update#setState idle
[main 2021-04-16T06:54:42.812Z] resolveShellEnv(): skipped (Windows)
(node:12488) electron: The default of contextIsolation is deprecated and will be changing from false to true in a future release of Electron.  See https://github.com/electron/electron/issues/23506 for more information
[main 2021-04-16T06:54:43.344Z] resolveShellEnv(): skipped (Windows)
[main 2021-04-16T06:54:43.358Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:54:43.360Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:54:43.569Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:54:44.110Z] SharedProcess: on vscode:createSharedProcessMessageChannel
[main 2021-04-16T06:54:44.110Z] resolveShellEnv(): skipped (Windows)
[main 2021-04-16T06:54:44.111Z] IPC Object URL: Registered new channel vscode:d43344eb-7bdd-4103-80e6-9af70497b3a1.
(node:12488) electron: The default of contextIsolation is deprecated and will be changing from false to true in a future release of Electron.  See https://github.com/electron/electron/issues/23506 for more information
[main 2021-04-16T06:54:44.235Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:54:44.238Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:54:44.299Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:54:44.304Z] resolveShellEnv(): skipped (Windows)
[main 2021-04-16T06:54:44.480Z] RequestService#request https://update.code.visualstudio.com/api/update/win32-x64-user/insider/25785b97e533b15ae3b12e48aaf4187416437e62
[main 2021-04-16T06:54:44.481Z] resolveShellEnv(): skipped (Windows)
[main 2021-04-16T06:54:44.501Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:54:44.689Z] SharedProcess: IPC ready
[main 2021-04-16T06:54:44.693Z] Main->SharedProcess#connect: connection established
[main 2021-04-16T06:54:44.742Z] SharedProcess: Overall ready
[main 2021-04-16T06:54:44.972Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:54:44.974Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:54:45.053Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:54:45.264Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:55:07.868Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:55:07.874Z] [File Watcher (node.js)] [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:55:08.083Z] [File Watcher (node.js)] >> normalized [CHANGED] c:\Users\etimokhov\AppData\Local\Programs\Microsoft VS Code Insiders\test\User\globalStorage
[main 2021-04-16T06:55:11.675Z] Lifecycle#window.on('close') - window ID 1
[main 2021-04-16T06:55:11.675Z] Lifecycle#unload() - window ID 1
[main 2021-04-16T06:55:11.681Z] Lifecycle#unload() - no veto (window ID 1)
[main 2021-04-16T06:55:11.726Z] StorageMainService: closed workspace storage (bf0fb64e91d1d2217f5a316b57076d19)
[main 2021-04-16T06:55:11.731Z] Lifecycle#onBeforeCloseWindow.fire() - window ID 1
[main 2021-04-16T06:55:11.754Z] IPC Object URL: Removed channel vscode:b9840501-6e4f-432f-b5c7-e7c055fc8b09.
[main 2021-04-16T06:55:11.754Z] Lifecycle#window.on('closed') - window ID 1
[main 2021-04-16T06:55:11.755Z] Lifecycle#onWillShutdown.fire()
[main 2021-04-16T06:55:11.756Z] IPC Object URL: Removed channel vscode:d43344eb-7bdd-4103-80e6-9af70497b3a1.
[main 2021-04-16T06:55:11.834Z] Lifecycle#app.on(window-all-closed)
[main 2021-04-16T06:55:11.835Z] Lifecycle#app.on(before-quit)
[main 2021-04-16T06:55:11.836Z] Lifecycle#onBeforeShutdown.fire()
[main 2021-04-16T06:55:11.837Z] [WindowsStateHandler] onBeforeShutdown {
  lastActiveWindow: {
    workspaceIdentifier: undefined,
    folder: 'file:///d%3A/projects/ts-transformer-properties-rename',
    backupPath: 'C:\\Users\\etimokhov\\AppData\\Local\\Programs\\Microsoft VS Code Insiders\\test\\Backups\\9c4c0b0188749bed5d377244339510ac',
    remoteAuthority: undefined,
    uiState: [Object: null prototype] {
      mode: 0,
      x: 448,
      y: 216,
      width: 1024,
      height: 768
    }
  },
  lastPluginDevelopmentHostWindow: undefined,
  openedWindows: []
}
[main 2021-04-16T06:55:11.853Z] Lifecycle#app.on(will-quit)
[main 2021-04-16T06:55:11.909Z] StorageMainService: closed global storage
```

**Comment by timocov** [comment]
I tried with folder on other drive (`D:` in my case instead of `C:` with Windows one) - the issue is still occurs.

**Comment by bpasero** [comment]
Ok thanks, unfortunately I am out of ideas.

**Comment by timocov** [comment]
Is it possible to debug vscode without building it locally? I can try to debug it if that works, but I think I'll need your help with some points where I need to start/pat attention on and so on.

**Comment by bpasero** [comment]
Thanks for the offer, https://github.com/microsoft/vscode/wiki/How-to-Contribute explains how to run out of sources. let me know if you need help getting there.

Btw the paths in question are not super long right? Because there is a limit

**Comment by timocov** [comment]
> Btw the paths in question are not super long right? Because there is a limit

Yeah, it's like `D:/projects/ts-transformer-properties-rename`.

> Thanks for the offer, microsoft/vscode/wiki/How-to-Contribute explains how to run out of sources. let me know if you need help getting there.

Cool, thanks! I'll get back here since I get any additional information/questions.

**Comment by timocov** [comment]
@bpasero do you have any advice where I can start debug from? Like where that list is generated or returned/set to OS?

**Comment by timocov** [comment]
Ok, it seems that I figure out where I need to start. I found this code (I don't use source maps):

```
try {
    y.app.setJumpList(f)
} catch (E) {
    this.logService.warn("updateWindowsJumpList#setJumpList", E)
}
```

which matches to this one https://github.com/microsoft/vscode/blob/94c9ea46838a9a619aeafb7e8afd1170c967bb55/src/vs/platform/workspaces/electron-main/workspacesHistoryMainService.ts#L386-L391

I added 2 breakpoints (and also enabled both "Pause on exceptions" and "Pause on caught exceptions"): one in `try` block and another one in `catch` block. It doesn't hit into `catch` block.

I tried to see what's in `f` variable:

```json
[
  {
    "type": "tasks",
    "items": [
      {
        "type": "task",
        "title": "New Window",
        "description": "Opens a new window",
        "program": "C:\\Users\\etimokhov\\AppData\\Local\\Programs\\Microsoft VS Code Insiders\\Code - Insiders.exe",
        "args": "-n",
        "iconPath": "C:\\Users\\etimokhov\\AppData\\Local\\Programs\\Microsoft VS Code Insiders\\Code - Insiders.exe",
        "iconIndex": 0
      }
    ]
  },
  {
    "type": "custom",
    "name": "Recent Folders",
    "items": [
      {
        "type": "task",
        "title": "ts-transformer-properties-rename",
        "description": "D:\\projects\\ts-transformer-properties-rename",
        "program": "C:\\Users\\etimokhov\\AppData\\Local\\Programs\\Microsoft VS Code Insiders\\Code - Insiders.exe",
        "args": "--folder-uri \"file:///d%3A/projects/ts-transformer-properties-rename\"",
        "iconPath": "explorer.exe",
        "iconIndex": 0
      }
    ]
  },
  {
    "type": "recent"
  }
]
```

and this doesn't work. But if I modify this list like this `f[0].items.push(...f[1].items)` it adds my recent folder to tasks list.

![image](https://user-images.githubusercontent.com/3112183/115020673-95903780-9ec3-11eb-8f31-af5605dc9dfc.png)

So probably the issue is with `custom` category.

**Comment by timocov** [comment]
I just tried to use example from https://www.electronjs.org/docs/api/app#appsetjumplistcategories-windows and it seems it does work correctly. This is what I got:

![image](https://user-images.githubusercontent.com/3112183/115021314-7c3bbb00-9ec4-11eb-8fe2-651025833d9f.png)

It seems that the only type="tasks" is working at the moment

**Comment by timocov** [comment]
As far as I understand `setJumpList` does to electron native API (which I cannot debug in devtools obviously). I've tried to read [the underlying code](https://github.com/electron/electron/blob/22a70eb8036825e3dc722473aa1de382e7371dff/shell/browser/ui/win/jump_list.cc#L225-L314) but didn't find anything specific there. Probably if it doesn't throw an error it's a issue of Windows itself?

Let me what you think about that.

**Comment by timocov** [comment]
Ok, I figured out what's going on here. Some time ago I've disabled "Show recently opened items in Jump Lists on Start or the taskbar and in File Explorer Quick Access" option in Windows Settings. I just enabled it and now it works fine! It isn't obvious that disabling this setting affects custom categories as well.

![image](https://user-images.githubusercontent.com/3112183/115026126-ca53bd00-9eca-11eb-8027-843618a95ae0.png)

Sorry for inconvenience.
